cmd/renew_certificate.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/dapr/cli/pkg/kubernetes"
+	"github.com/dapr/cli/pkg/print"
+	"github.com/dapr/cli/utils"
+)
+
+var (
+	certificatePasswordFile     string
+	caRootCertificateFile       string
+	issuerPrivateKeyFile        string
+	issuerPublicCertificateFile string
+	validUntil                  int
+	restartDaprServices         bool
+)
+
+var RenewCertificateCmd = &cobra.Command{
+	Use:   "renew-cert",
+	Short: "Rotates Dapr root certificate of your kubernetes cluster",
+	Example: `
+# Generates new root and issuer certificates for kubernetest cluster
+dapr mtls renew-cert -k --valid-until <no of days> --restart true
```

Comment:
Other mtls commands don't seem to support the `-k` flag. Is that expected? 

```bash
dapr mtls expiry -k
Error: unknown shorthand flag: 'k' in -k
```

Reply:
this is expected in case we want to enhance it for standalone.
`export` and `expiry` does not make much sense for standalone mode imo.
In general, I think it should be with the specific subcommands whether -k flag is needed or not.

Reply:
If `mtls` only supports k8s, then the subcommands should also enforce it ideally. Instead of silently failing, we should complain that it's a required flag and only k8s mode is supported. This can be tracked separately, but I think we should do it. Thoughts? 

/cc @mukundansundar

Reply:
I will be adding `RenewCertificateCmd.MarkFlagRequired("kubernetes")` in new commit.
We can enforce this flag through all subcommands but it will change the current behavior from  `dapr mtls expiry` to of `dapr mts expiry -k` 
@mukundansundar